yarn/src/main/scala/org/apache/spark/deploy/yarn/ClientArguments.scala

```diff
@@ -87,6 +92,21 @@ private[spark] class ClientArguments(args: Array[String], sparkConf: SparkConf)
       throw new IllegalArgumentException(
         "You must specify at least 1 executor!\n" + getUsageMessage())
     }
+    if (isClusterMode) {
+      for (key <- Seq(amMemKey, amMemOverheadKey)) {
+        if (sparkConf.getOption(key).isDefined) {
+          println(s"$key is set but does not apply in cluster mode.")
```

Comment:
As `ClientArguments.scala` didn't extends Logging class, only `println` can be used here.
Yep, if user set the config values that never be used in that mode, we should give a prompt.

BTW, `spark.driver.memory` is used in both modes, so I deleted the meesage about it.